src/main/java/core/basesyntax/Pair.java

```diff
@@ -1,5 +1,45 @@
 package core.basesyntax;
 
-class Pair {
+class Pair<T, V> {
+    private T first;
+    private V second;
 
+    private Pair(T first, V second) {
+        this.first = first;
+        this.second = second;
+    }
+
+    public static <T, V> Pair<T, V> of(T first, V second) {
+        return new Pair<>(first, second);
+    }
+
+    @Override
+    public boolean equals(Object o) {
+        if (this == o) {
+            return true;
+        }
+        if (o == null || getClass() != o.getClass()) {
+            return false;
+        }
+        Pair<T, V> pair = (Pair<T, V>) o;
+        return (first == null && pair.first == null && second == null && pair.second == null)
+                || (first != null && first.equals(pair.first)
+                && second != null && second.equals(pair.second));
```

Comment:
this looks a bit wrong

let's recall how we compared fields of the `rectangle` objects